pkg/trace/agent/agent.go

```diff
@@ -349,6 +355,23 @@ func newChunksArray(chunks []*pb.TraceChunk) []*pb.TraceChunk {
 
 var _ api.StatsProcessor = (*Agent)(nil)
 
+// filterSpans removes all spans for which the provided FilterSpan function returns true
+func (a *Agent) filterSpans(p *api.Payload) {
+	if a.FilterSpan == nil {
+		return
+	}
+	for i := 0; i < len(p.Chunks()); i++ {
+		chunk := p.Chunk(i)
```

Comment:
Why not this?
```suggestion
	for _, chunk := range p.Chunks() {
```

Reply:
I'm guessing you mean `for i, chunk := ..`? It seems like `p.Chunks` takes in an index. I'll definitely refactor to use this, good suggestion!